src/spi2_pool.rs

```diff
@@ -0,0 +1,211 @@
+use crate::gpio::{AnyOutputPin, InputPin, Output, OutputPin, PinDriver};
+use crate::peripheral::{Peripheral, PeripheralRef};
+use crate::spi2::{config, SpiBusMasterDriver, SpiMasterDriver};
+
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use core::ptr;
+
+use esp_idf_sys::*;
+
+// TODO -> custom error that wraps around esperror
+// cs default high low setting -
+// needed to makte SpiBusMasterDriver fields pub because different module
+// needed to make SpiBusMasterDriver flush fn pub
+
+// performance measurment on esp32c3 -> 
+// debug build: 50 uS deley between cs low -> first clk out
+// release build : 15 uS deley between cs low -> first cll out
+
+// debug build: 4uS delay between last clk out -> cs pin high
+// release build: 3uS delay between last clk out -> cs pin high
+
+#[cfg(not(esp32c3))]
+const ESP_MAX_SPI_DEVICES: usize = 3;
+#[cfg(esp32c3)]
+const ESP_MAX_SPI_DEVICES: usize = 6;
+
+pub struct SpiConfigPool<'d, T> {
+    shared_handles: HashMap<u32, spi_device_handle_t>,
```

Comment:
HashMap is an overkill, unless you plan to use tens to hundreds of devices here. And requires STD. Why not a `BTreeMap` or even a simple array of `(u32, spi_device_handle_t)` here?